src/AuthenticationService.php

```diff
@@ -428,4 +429,50 @@ public function getLoginRedirect(ServerRequestInterface $request): ?string
 
         return $parsed['path'] . $parsed['query'];
     }
+
+    /**
+     * Impersonates a user
+     *
+     * @param ServerRequestInterface $request
+     * @param ResponseInterface $response
+     * @param \ArrayAccess|array $impersonator
+     * @param \ArrayAccess|array $impersonated
+     * @return \ArrayAccess|array
+     */
+    public function impersonate(ServerRequestInterface $request, ResponseInterface $response, \ArrayAccess|array $impersonator, \ArrayAccess|array $impersonated): \ArrayAccess|array
+    {
+        /** @var PersistenceInterface $provider */
+        $provider = $this->getAuthenticationProvider();
+        if (!($provider instanceof PersistenceInterface)) {
+            throw new \InvalidArgumentException('Provider must implement PersistenceInterface to be able to impersonate user.');
+        }
+        return $provider->impersonate($request, $response, $impersonator, $impersonated);
+    }
+
+    /**
+     * Stops impersonation
+     *
+     * @param ServerRequestInterface $request
+     * @param ResponseInterface $response
+     * @return \ArrayAccess|array
+     */
+    public function stopImpersonate(ServerRequestInterface $request, ResponseInterface $response): \ArrayAccess|array
```

Comment:
Should this method be `stopImpersonating()` or `stopImpersonation()`?

Reply:
Done